pkg/storage/system_intake_contract_number.go

```diff
@@ -0,0 +1,99 @@
+package storage
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+	"github.com/lib/pq"
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/models"
+	"github.com/cmsgov/easi-app/pkg/sqlqueries"
+)
+
+// SetSystemIntakeContractNumbers links given Contract Numbers to given System Intake ID
+func (s *Store) SetSystemIntakeContractNumbers(ctx context.Context, tx *sqlx.Tx, systemIntakeID uuid.UUID, contractNumbers []string) error {
```

Comment:
Instead of explicitly passing a tx, it could  be more ideal to pass a sqlutils named preparer. (See CreateTRBRequestForm).

This way, you could pass a transaction, or a store.

Reply:
I don't think we _ever_ want to pass anything _but_ a transaction here, since the calls in this function are heavily reliant on each other. 

Forcing a `tx` here is a bit different from some of the other store methods, but it prevents someone calling this outside a transaction, which isn't what we want

Reply:
Perhaps adding a comment to this function clarifying that could be useful

```suggestion
// This function opts to take a *sqlx.Tx instead of a NamedPreparer because the SQL calls inside this function are heavily intertwined, and we never want to call them outside the scope of a transaction
func (s *Store) SetSystemIntakeContractNumbers(ctx context.Context, tx *sqlx.Tx, systemIntakeID uuid.UUID, contractNumbers []string) error {
```

Reply:
Nice! Thank you both!